src/delta/delta.c

```diff
@@ -340,23 +345,29 @@ static int enumerate_dir(
 
         STRV_FOREACH(t, files) {
                 _cleanup_free_ char *p = NULL;
+                char *bp = NULL;
 
                 p = path_join(path, *t);
                 if (!p)
                         return -ENOMEM;
 
-                log_debug("Adding at top: %s %s %s", basename(p), special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
-                r = ordered_hashmap_put(top, basename(p), p);
+                r = path_extract_filename(p, &bp);
+                if (r < 0)
+                        return r;
+
+                log_debug("Adding at top: %s %s %s", bp, special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
+                r = ordered_hashmap_put(top, bp, p);
                 if (r >= 0) {
                         p = strdup(p);
                         if (!p)
                                 return -ENOMEM;
-                } else if (r != -EEXIST)
+                } else if (r != -EEXIST) {
                         return r;
+                }
 
-                log_debug("Adding at bottom: %s %s %s", basename(p), special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
-                free(ordered_hashmap_remove(bottom, basename(p)));
-                r = ordered_hashmap_put(bottom, basename(p), p);
+                log_debug("Adding at bottom: %s %s %s", bp, special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), p);
+                free(ordered_hashmap_remove(bottom, bp));
+                r = ordered_hashmap_put(bottom, bp, p);
```

Comment:
Whenever hashmap is involved, the hash_ops should be changed instead, so that the key is also automatically destructed when ordered_hashmap_free(). Refer to https://github.com/systemd/systemd/pull/32603/commits/0bb4614adc68c4d834db3a3090064eef9fede0b3 as an example.